salt/cloud/clouds/clc.py

```diff
@@ -0,0 +1,368 @@
+        queue = clc.v1.Blueprint.GetStatus(request_id=(req_id))
+        if (queue["PercentComplete"] == 100):
+            
+            server_name = queue["Servers"][0]
+            creds = get_creds()
+            clc.v2.SetCredentials(creds["user"],creds["password"]) 
+            ip_addresses = clc.v2.Server(server_name).ip_addresses
+            internal_ip_address = ip_addresses[0]["internal"] 
+            return internal_ip_address            
+        else:
+            counter = counter + 1
+            log.info("Creating Cloud VM " + nodename + " Time out in " + str(10 - counter) + " minutes")
+            time.sleep(60)
+def create(vm_):
+    creds = get_creds()
+    clc.v1.SetCredentials(creds["token"],creds["token_pass"])
+    cloud_profile = config.is_provider_configured(
+        __opts__,
+        __active_provider_name__ or __virtualname__,
+        ('token',)
+    )
+    group = config.get_cloud_config_value(
+        'group', vm_, __opts__, search_global=False, default=None,
+    )
+    name = vm_['name']
+    description = config.get_cloud_config_value(
+        'description', vm_, __opts__, search_global=False, default=None,
+    )
+    ram = config.get_cloud_config_value(
+        'ram', vm_, __opts__, search_global=False, default=None,
+    )
+    backup_level = config.get_cloud_config_value(
+        'backup_level', vm_, __opts__, search_global=False, default=None,
+    )
+    template = config.get_cloud_config_value(
+        'template', vm_, __opts__, search_global=False, default=None,
+    )
+    password = config.get_cloud_config_value(
+        'password', vm_, __opts__, search_global=False, default=None,
+    )
+    cpu = config.get_cloud_config_value(
+        'cpu', vm_, __opts__, search_global=False, default=None,
+    )
+    network = config.get_cloud_config_value(
+        'network', vm_, __opts__, search_global=False, default=None,
+    )
+    location = config.get_cloud_config_value(
+        'location', vm_, __opts__, search_global=False, default=None,
+    )
+    if len(name) > 6:
+        name = name[0:6]
+    if len(password) < 9:
+        password = ''
+    clc_return = clc.v1.Server.Create(alias=None,location=(location),name=(name),template=(template),cpu=(cpu),ram=(ram),backup_level=(backup_level),group=(group), network=(network),description=(description),password=(password))
+    req_id = clc_return["RequestID"]
+    vm_['ssh_host'] = get_build_status(req_id,name)
+    __utils__['cloud.fire_event'](
+        'event',
+        'waiting for ssh',
+        'salt/cloud/{0}/waiting_for_ssh'.format(name),
+        sock_dir=__opts__['sock_dir'],
+        args={'ip_address': vm_['ssh_host']},
+        transport=__opts__['transport']
+    )
+
+    # Bootstrap!
+    ret = __utils__['cloud.bootstrap'](vm_, __opts__)
+    return_message = {"Server Name":name,"IP Address":vm_['ssh_host']}
+    ret.update(return_message)
+    return(return_message)
+
+def destroy(name, call=None):
+    print("destroying ")
```

Comment:
Please don't print from here. We need to return True instead, most likely. We could also log if desired.